lib/utils/lookupObjectUtil.js

```diff
@@ -0,0 +1,96 @@
+const { messages } = require('elasticio-node');
+const { Readable } = require('stream');
+const { LookupObjectById } = require('@elastic.io/oih-standard-library/lib/actions/lookupObject');
+const attachments = require('../attachments');
+
+const MAX_FILE_SIZE = 104857600; // 100 MiB
+
+class SftpLookupObject extends LookupObjectById {
+  constructor(logger, client) {
+    super(logger);
+    this.client = client;
+  }
+
+  // eslint-disable-next-line class-methods-use-this, no-unused-vars
+  getId(msg, cfg) {
+    return msg.body.filename;
+  }
+
+  async lookupObject(filename, cfg) {
+    const { directory } = cfg;
+    const dir = directory.substring(directory.length - 1) === '/'
+      ? directory.substring(0, directory.length - 1)
+      : directory;
+    this.logger.info(`Starting verify is file with name: ${filename} is exists in directory ${dir}`);
+    const list = await this.client.list(dir, new RegExp(filename));
+    const files = list.filter((file) => file.name === filename && file.type === '-');
+    if (files.length !== 1) {
+      if (files.length === 0) {
+        this.logger.info(`File with name ${filename} was not found`);
+      } else {
+        this.logger.info('More than one file were found');
+      }
+      return null;
+    }
+    const [file] = files;
+    const { size } = file;
+    if (size > MAX_FILE_SIZE) {
+      throw new Error(`File with filename: ${filename} has size = ${size} MiB and can not be processed.`);
+    }
+    this.logger.info(`File with name: ${filename} is exists in directory ${dir}`);
+    const filePath = `${dir}/${filename}`;
+    this.logger.info(`Reading ${filePath} into read stream`);
+    const buffer = await this.client.get(filePath);
+    const readStream = new Readable();
+    readStream.push(buffer);
+    readStream.push(null);
+    const result = messages.newMessageWithBody({
+      filename,
+      size,
+    });
+    this.logger.info('Starting add attachment to result');
+    await attachments.addAttachment.call(this, result, filename, readStream, size);
+    this.logger.info('Attachment successfully added to message');
+    await this.client.end();
```

Comment:
Better move it to action `process` function - there are some cases here when connection close is not executing